onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -3,7 +3,7 @@
 
 #include "core/common/inlined_containers.h"
 #include "core/providers/shared_library/provider_api.h"
-#include <core/platform/env_var_utils.h>
+#include "core/platform/env_var_utils.h"
```

Comment:
move this line before "core/providers/shared_library/provider_api.h" ?

Reply:
Impossible!

`provider_api.h` is the evil header. It triggers ODR violation if not ordered correctly.